File: services/api-agente/src/infrastructure/dataprovider/usuario_data_provider.py
import uuid

from src.config.database import SessionLocal
from src.infrastructure.entity.usuario_entity import UsuarioEntity
from src.infrastructure.exceptions.data_provider_exception import DataProviderException
from src.infrastructure.mapper.usuario_mapper import UsuarioMapper
import logging

logger = logging.getLogger(__name__)

class UsuarioDataProvider:

    def __init__(self, usuario_mapper: UsuarioMapper):
        self.usuario_mapper = usuario_mapper

    def consultar_por_id(self, id: str):
        session = SessionLocal()

        try:
            uuid_bytes = uuid.UUID(id).bytes
            entity = session.query(UsuarioEntity).filter(UsuarioEntity.id == uuid_bytes).first()
            return self.usuario_mapper.para_domain(entity)
        except Exception as e:
            session.rollback()
            logger.exception("Erro ao consultar usuário pelo id. %s", e)
            raise DataProviderException("Erro ao consultar usuário pelo id.")
        finally:
            session.close()

File: services/api-intermediaria/src/main/java/com/gumeinteligencia/api_intermidiaria/infrastructure/dataprovider/UsuarioDataProvider.java
package com.gumeinteligencia.api_intermidiaria.infrastructure.dataprovider;

import com.gumeinteligencia.api_intermidiaria.application.gateways.UsuarioGateway;
import com.gumeinteligencia.api_intermidiaria.domain.Usuario;
import com.gumeinteligencia.api_intermidiaria.infrastructure.exceptions.DataProviderException;
import com.gumeinteligencia.api_intermidiaria.infrastructure.mapper.UsuarioMapper;
import com.gumeinteligencia.api_intermidiaria.infrastructure.repository.UsuarioRepository;
import com.gumeinteligencia.api_intermidiaria.infrastructure.repository.entity.UsuarioEntity;
import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.springframework.stereotype.Component;

import java.util.Optional;

@Component
@Slf4j
@RequiredArgsConstructor
public class UsuarioDataProvider implements UsuarioGateway {

    private final UsuarioRepository repository;
    private final String MENSAGEM_ERRO_CONSULTAR_POR_TELEFONE_CONECTADO = "Erro ao consultar usuário pelo telefone conectado.";

    @Override
    public Optional<Usuario> consultarPorTelefoneConectado(String telefoneConectado) {
        Optional<UsuarioEntity> usuarioEntity;

        try {
            usuarioEntity = repository.findByTelefoneConectado(telefoneConectado);
        } catch (Exception ex) {
            log.error(MENSAGEM_ERRO_CONSULTAR_POR_TELEFONE_CONECTADO, ex);
            throw new DataProviderException(MENSAGEM_ERRO_CONSULTAR_POR_TELEFONE_CONECTADO, ex.getCause());
        }

        return usuarioEntity.map(UsuarioMapper::paraDomain);
    }
}
